### PR TITLE
[numpy] fix op repeat with list input

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
#16357 repeat

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by Yiyan66** [comment]
@mxnet-bot run ci [windows-gpu]

File: src/operator/numpy/np_repeat_op-inl.h [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    return true;
+  }
+
+  // If repeats > 0, multiply the size of the corresponding axis by repeats
+  if (static_cast<bool>(axisOpt)) {
+    int ndims = ishape.ndim();
+    int axis = axisOpt.value();
+    if (axis < 0) {
+      axis += ndims;
+    }
```
Comment (yzhliu) [inline src/operator/numpy/np_repeat_op-inl.h:108]: can we unify the logic with those in `GetRepeatsParams`

File: python/mxnet/symbol/numpy/_symbol.py [hunk 2]
```diff
@@ -2446,7 +2446,15 @@ def repeat(a, repeats, axis=None):
...
-    return _npi.repeat(a, repeats=repeats, axis=axis)
+    if isinstance(repeats, numeric_types):
+        return _npi.repeat(a, repeats=repeats, axis=axis)
+    if len(repeats) == 1:
+        return _npi.repeat(a, repeats[0], axis)
+    if axis is not None:
+        tmp = swapaxes(a, 0, axis)
+        res = _npi.repeats(tmp, repeats=repeats, axis=0)
+        return swapaxes(res, 0, axis)
+    return _npi.repeats(a, repeats=repeats, axis=axis)
```
Comment (yzhliu) [inline python/mxnet/symbol/numpy/_symbol.py:2457]: can we unify these two ops? numeric_type can be passed as `[repeats]`

File: src/operator/numpy/np_repeat_op-inl.h [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    SHAPE_ASSIGN_CHECK(*out_attrs, 0, shape);
+  }
+  return shape_is_known(out_attrs->at(0));
+}
+
+namespace {  // unnamed namespace to keep scope of the struct within the file
+struct RepeatInOutIndex {
+  int index[50];
+};
+}  // unnamed namespace
```
Comment (yzhliu) [inline src/operator/numpy/np_repeat_op-inl.h:126]: seems to be an unused struct?

File: python/mxnet/ndarray/numpy/_op.py [hunk 4]
```diff
@@ -4076,7 +4076,13 @@ def repeat(a, repeats, axis=None):
            [3, 4],
            [3, 4]])
     """
-    return _api_internal.repeat(a, repeats, axis)
+    if isinstance(repeats, numeric_types):
+        repeats = [repeats]
+    if axis is not None:
+        tmp = swapaxes(a, 0, axis)
+        res = _api_internal.repeats(tmp, repeats, 0)
```
Comment (yzhliu) [inline python/mxnet/ndarray/numpy/_op.py:4083]: can we unify .repeats -> .repeat?

**Comment by yzhliu** [comment]
Thanks @Yiyan66

**Pull request merged**